### ISSUE TITLE
[BUG] Reliable network packages are not reliable enough

### ISSUE THREAD

**Issue by o01eg** [amethyst/amethyst#1774]
## Description

I'm trying to use network and found the example server skips reliable packages sent by the example client.

## Reproduction Steps

1. Change `net_client` code to send some count of packages once.
2. Change type of messages from `unreliable` to `reliabe_unordered`.
3. Build `net_client` and `net_server` examples.
4. Launch `net_server`.
5. Launch `net_client`.
6. Switch to the serve console and find packages weren't read from start or even weren't read entirely.

## What You Expected to Happen

Each `reliabe_unordered` package was read by the server.

## What Actually Happened

The server starts to read packages from some time skipping all packages before.

## Screenshots and Logs

If I set client to send 1000 packages:

```
$ ./target/debug/examples/net_server
[INFO][amethyst::app] Initializing Amethyst...
[INFO][amethyst::app] Version: 0.11.1
[INFO][amethyst::app] Platform: x86_64-unknown-linux-gnu
[INFO][amethyst::app] Amethyst git commit: 94830cb1a930d15ab62749c23cdbb550f2a9fa79
[INFO][amethyst::app] Rustc version: 1.37.0-dev Dev
[INFO][amethyst::app] Rustc git commit: 8ec39423dd0d02f81866b675dd9299972e1b9af5
Received 0 messages this frame connections: 0
Received 0 messages this frame connections: 0
Received 0 messages this frame connections: 0
Received 0 messages this frame connections: 1
[INFO][net_server] CL: frame:0,abs_time:0,c:640
[INFO][net_server] CL: frame:0,abs_time:0,c:641
[INFO][net_server] CL: frame:0,abs_time:0,c:642
[INFO][net_server] CL: frame:0,abs_time:0,c:643
[INFO][net_server] CL: frame:0,abs_time:0,c:644
[INFO][net_server] CL: frame:0,abs_time:0,c:645
[INFO][net_server] CL: frame:0,abs_time:0,c:646
[INFO][net_server] CL: frame:0,abs_time:0,c:647
[INFO][net_server] CL: frame:0,abs_time:0,c:648
...
[INFO][net_server] CL: frame:0,abs_time:0,c:991
[INFO][net_server] CL: frame:0,abs_time:0,c:992
[INFO][net_server] CL: frame:0,abs_time:0,c:993
[INFO][net_server] CL: frame:0,abs_time:0,c:994
[INFO][net_server] CL: frame:0,abs_time:0,c:995
[INFO][net_server] CL: frame:0,abs_time:0,c:996
[INFO][net_server] CL: frame:0,abs_time:0,c:997
[INFO][net_server] CL: frame:0,abs_time:0,c:998
[INFO][net_server] CL: frame:0,abs_time:0,c:999
Received 360 messages this frame connections: 1
```

## Your Environment

1. Operating System: Gentoo Linux AMD64
2. Operating System Version: 17.1
3. Version of Amethyst: master  94830cb1a930d15ab62749c23cdbb550f2a9fa79

**Comment by jstnlef** [comment]
We, the network team, are currently working on a full revamp of the networking code provided by the engine. Part of that will include more interesting examples demonstrating the reliable use case. The reason you are seeing strange behavior has to do with the way the underlying reliability layer (laminar) works. In order to ensure reliability, the client and the server must continue to exchange messages with one another or else the protocol will exhibit errant behavior.

**Comment by spacejam** [comment]
@jstnlef where can I find more info about those efforts? I'm curious from a reliable distributed systems perspective, and might be able to lend some help with testing, depending on the specifics of the architecture. This is the only issue currently open tagged `team: network`

**Comment by jstnlef** [comment]
I'm in the middle of putting together a PR from my fork of amethyst. You can see the progress here: https://github.com/jstnlef/amethyst/tree/rewrite-the-network-library. I've been "away from computer" for a bit now but I'm hoping to ramp up the progress again this week.

**Comment by o01eg** [comment]
I suppose it could be closed as TCP protocol support was added in #1917
